engine/lite/components-api/src/main/scala/pl/touk/nussknacker/engine/lite/api/customComponentTypes.scala

```diff
@@ -25,11 +25,20 @@ object customComponentTypes {
       capabilityTransformer: CapabilityTransformer[F]
   )
 
+  object LiteSource {
+    val DefaultTraceIdHeader: String = "trace-id"
+    val EmptyHeaders                 = Map.empty[String, String]
+  }
+
   trait LiteSource[Input] extends Source {
 
+    // protected for overriding purposes
+    protected def extractTraceId(headers: Map[String, String]): Option[TraceId] =
+      headers.get(LiteSource.DefaultTraceIdHeader).map(TraceId(_))
+
     def createTransformation[F[_]: Monad](
         evaluateLazyParameter: CustomComponentContext[F]
-    ): Input => ValidatedNel[ErrorType, Context]
+    ): (Input, Map[String, String]) => ValidatedNel[ErrorType, Context]
```

Comment:
We should replace `Input` with some smarter type in `RequestResponseSource` instead of changing this on this level. Take a look that in `LiteKafkaSource`, headers has no sense because we already use low-level entity - `ConsumerRecord` which contains these headers. We should follow the same approach in R-R